app/models/ext_management_system.rb

```diff
@@ -84,11 +84,16 @@ def self.supported_types_and_descriptions_hash
   def hostname_uniqueness_valid?
     return unless hostname_required?
     return unless hostname.present? # Presence is checked elsewhere
-    # check uniqueness per provider type
-
-    existing_hostnames = (self.class.all - [self]).map(&:hostname).compact.map(&:downcase)
-
-    errors.add(:hostname, N_("has to be unique per provider type")) if existing_hostnames.include?(hostname.downcase)
+    hostname_without_slash = hostname.downcase.gsub(/\/$/, '')
```

Comment:
You're doing `hostname.downcase.gsub()` in a lot of places, can you add a `normalized_hostname` method that does this in one place?